### DISCUSSION TITLE
vue-query hooks can only be used inside setup() function

### DISCUSSION THREAD

**Discussion by khaledOghli** [TanStack/query#5178] [Q&A]
Hi Guys 👍🏻 
How Can I use " useQuery " to get data inside method in setup.!?
`
import useQuery from "~/composables/useFetchQuery";

const getEmittedData = (taskID, toggleInfo) => {
  if (toggleInfo) {
        const currentHistoryRow = data.value.newHistories.filter((item) => item.taskID === taskID);
        currentTaskID.value = taskID;
        const { isLoading } = useQuery.fetchQuery(
          [`historyFeedback-${taskID}`],
          getHistoryFeedback,
          onHistoryFeedbackSuccess,
          onHistoryFeedbackError
        );
        currentHistoryRow['isLoading'] = isLoading;
  }
};

`
Uncaught Error: vue-query hooks can only be used inside setup() function.

**Comment by DamianOsipiuk** [comment]
Use `computed` value to control when query will run. And set the value from within your callback.
Ex:
```ts
const isEnabled = computed(() => Boolean(currentTaskID.value))

const query = useQuery({
  queryKey,
  queryFn,
  // query will start loading when `currentTaskID` is set
  enabled: isEnabled
})

**Reply by bennyz327** [reply]
I’m having trouble getting the enabled option in useQuery to work. If isEnabled is initially set to false when the component mounts and later switched to true, the queryFn still never runs to fetch data.
